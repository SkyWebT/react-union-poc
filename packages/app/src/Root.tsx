import './Root.css';

import React from 'react';
import { Union } from 'react-union';

import routes from './routes';

const Root = () => <Union routes={routes} />;

export default Root;
