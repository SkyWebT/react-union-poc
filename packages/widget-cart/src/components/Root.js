import { useSelector, useDispatch } from 'react-redux';
import React from 'react';

const getSelected = state => Object.values(state.products).filter(p => p.selected);
const getTotal = state => {
	const products = getSelected(state);
	return products.reduce((acc, cur) => {
		acc += cur.price;
		return acc;
	}, 0);
};
const Root = () => {
	const { products, total } = useSelector(state => ({ products: getSelected(state), total: getTotal(state) }));
	const d = useDispatch();

	const removeFromCart = id => {
		d({
			type: 'TOGGLE',
			payload: { id },
		});
	};
	return (
		<div style={{ border: '1px solid grey', padding: 20, width: '40vw' }}>
			<h2>Cart </h2>
			{products.length > 0 ? (
				products.map(({ id, name, price }) => (
					<div key={id} style={{ display: 'flex', justifyContent: 'space-between' }}>
						<h3>
							{name} - ${price}
						</h3>
						<div>
							<button onClick={() => removeFromCart(id)}>remove</button>
						</div>
					</div>
				))
			) : (
				<div> empty </div>
			)}
			<h3>Total: ${total}</h3>
		</div>
	);
};

export default Root;
