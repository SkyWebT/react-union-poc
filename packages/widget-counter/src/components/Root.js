import { useSelector, useDispatch } from 'react-redux';
import React from 'react';
import { actions } from '@skytv/store';

const Root = () => {
	const { counter } = useSelector(s => s.counter);
	const d = useDispatch();
	const inc = () => {
		d(actions.counter.increment());
	};
	const dec = () => {
		d(actions.counter.decrement());
	};
	return (
		<div style={{ border: '1px solid grey', margin: 20, padding: 20 }}>
			<div>{counter}</div>

			<button onClick={inc}>+</button>
			<button onClick={dec}>-</button>
		</div>
	);
};

export default Root;
