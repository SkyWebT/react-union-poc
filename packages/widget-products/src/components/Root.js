import { useSelector, useDispatch } from 'react-redux';
import React from 'react';
import { actions } from '@skytv/store';
const Root = () => {
  const products = useSelector(s => s.products);
  const dispatch = useDispatch();
  console.log(products);
  const addToCart = id => {
    dispatch(actions.prodducts.toggle(id));
  };
  return (
    <div style={{ border: '1px solid grey', padding: 20, width: '40vw' }}>
      <h2>Product list</h2>

      {Object.values(products).map(({ id, name, price, selected }) => (
        <div key={id} style={{ display: 'flex', justifyContent: 'space-between' }}>
          <h3>
            {name} - ${price}
          </h3>
          <div>
            <button disabled={selected} onClick={() => addToCart(id)}>
              buy
            </button>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Root;
