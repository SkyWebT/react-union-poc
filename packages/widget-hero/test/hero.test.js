import React from 'react';
import Root from '../src/components/Root';
import { render } from '@testing-library/react';

describe('<Root />', () => {
  const requiredProps = {};

  it('should match snapshot', () => {
    const { asFragment } = render(<Root {...requiredProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
