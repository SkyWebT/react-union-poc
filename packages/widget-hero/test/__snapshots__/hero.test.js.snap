// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should match snapshot 1`] = `
<DocumentFragment>
  <img
    src="logo.png"
  />
</DocumentFragment>
`;
